### PR TITLE
Add dark mode

### LINKED ISSUE #40: Night mode detection
If we provide a night mode/dark theme for the new frontend it would be fancy to use the CSS based detection which would allow browsers to indicate that users prefer dark themes and CodiMD could pick that up automatically.

Working code for the old frontend exists in this PR of mine: https://github.com/codimd/server/pull/339

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
- The overall application theme
- Dark mode toggle button

### Description
This PR adds a dark mode (theme). Its state will be stored and loaded from localStorage, but the initial state will be determined based on the preferred browser color scheme.

**Please test this PR out on https://ui-test-dark.hedgedoc.net and report problems here.**

### Steps

- [ ] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
Fixes #40

### CODE REVIEW

**Comment by ErikMichelson** [comment]
Things to fix:
 - sync-scroll button
 - x-overflow of codemirror
 - font-color of banner

**Comment by DerMolly** [comment]
grafik

**Comment by DerMolly** [comment]
light
grafik
grafik
dark

File: src/components/markdown-renderer/markdown-renderer.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import '../../../node_modules/github-markdown-css/github-markdown.css';
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/markdown-renderer.scss:1]: We should drop the external package completely.

File: src/style/dark.scss [hunk 2]
```diff
@@ -0,0 +1,150 @@
+// Darkly 4.5.2
+// Bootswatch
```
Comment (ErikMichelson) [inline src/style/dark.scss:2]: Although we used that file as template, it probably has not too much in common with the result. Could this reference therefore be deleted?

Comment (DerMolly) [reply src/style/dark.scss]: done

**Comment by DerMolly** [comment]
the toolbar with no special color in the dark mode.
grafik

**Comment by DerMolly** [comment]
dropdown background color
grafik

**Review by mrdrogdrog — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**